subxt/src/backend/rpc/reconnecting_rpc_client/mod.rs

```diff
@@ -0,0 +1,630 @@
+        self.headers = headers;
+        self
+    }
+
+    /// Configure which retry policy to use when a connection is lost.
+    ///
+    /// Default: Exponential backoff 10ms
+    pub fn retry_policy<T>(self, retry_policy: T) -> ClientBuilder<T> {
+        ClientBuilder {
+            max_request_size: self.max_request_size,
+            max_response_size: self.max_response_size,
+            retry_policy,
+            #[cfg(feature = "native")]
+            ping_config: self.ping_config,
+            #[cfg(feature = "native")]
+            headers: self.headers,
+            max_redirections: self.max_redirections,
+            max_log_len: self.max_log_len,
+            id_kind: self.id_kind,
+            max_concurrent_requests: self.max_concurrent_requests,
+            request_timeout: self.request_timeout,
+            connection_timeout: self.connection_timeout,
+        }
+    }
+
+    #[cfg(feature = "native")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "native")))]
+    /// Configure the WebSocket ping/pong interval.
+    ///
+    /// Default: 30 seconds.
+    pub fn enable_ws_ping(mut self, ping_config: PingConfig) -> Self {
+        self.ping_config = Some(ping_config);
+        self
+    }
+
+    #[cfg(feature = "native")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "native")))]
+    /// Disable WebSocket ping/pongs.
+    ///
+    /// Default: 30 seconds.
+    pub fn disable_ws_ping(mut self) -> Self {
+        self.ping_config = None;
+        self
+    }
+
+    /// Build and connect to the target.
+    pub async fn build(self, url: String) -> Result<Client, RpcError> {
+        let (tx, rx) = mpsc::unbounded_channel();
+        let client = Retry::new(self.retry_policy.clone(), || {
+            platform::ws_client(url.as_ref(), &self)
+        })
+        .await?;
+        let (reconn_tx, reconn_rx) = reconnect_channel();
+
+        platform::spawn(background_task(client, rx, url, reconn_tx, self));
+
+        Ok(Client {
+            tx,
+            reconnect: reconn_rx,
+        })
+    }
+}
+
+impl Client {
+    /// Perform a JSON-RPC method call.
+    pub async fn request(
+        &self,
+        method: String,
+        params: Option<Box<RawValue>>,
+    ) -> Result<Box<RawValue>, Error> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(Op::Call {
+                method,
+                params: RpcParams(params),
+                send_back: tx,
+            })
+            .map_err(|_| Error::Dropped)?;
+
+        rx.await.map_err(|_| Error::Dropped)?
+    }
+
+    /// Perform a JSON-RPC subscription.
+    pub async fn subscribe(
+        &self,
+        subscribe_method: String,
+        params: Option<Box<RawValue>>,
+        unsubscribe_method: String,
+    ) -> Result<Subscription, Error> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(Op::Subscription {
+                subscribe_method,
+                params: RpcParams::new(params),
```

Comment:
Nit: inconsistent: we use `RpcParams(params)` above and `RpcParams::new(params)` here :)

Maybe we can remove RpcParams entirely to save a few lines or remove the `new` fn?